### PR TITLE
changed rootfinders to return also when absolute tolerance is less th…

### PR DESCRIPTION
**Pull request by hnil**
changed rootfinders to return when also when abs tol is less than tol.
In some cases the solver would not report convergence even if interval was 1e-17 when it was
solved for saturations.

### CODE REVIEW

File: opm/common/utility/numeric/RootFinders.hpp [hunk 1]
```diff
@@ -124,7 +124,7 @@ namespace Opm
             iterations_used = 0;
             // In every iteraton, x1 is the last point computed,
             // and x0 is the last point computed that makes it a bracket.
-            while (fabs(x1 - x0) >= 1e-9*eps) {
+            while (fabs(x1 - x0) >= eps) {
```
Comment (blattms) [inline opm/common/utility/numeric/RootFinders.hpp:127]: Seems like quite a drastic change in tolerance.

Comment (hnil) [reply opm/common/utility/numeric/RootFinders.hpp:127]: The original tolerance was so normally 1e-19 which mean that it was never trigged...

Comment (totto82) [reply opm/common/utility/numeric/RootFinders.hpp:127]: Note that the RegulaFalsiBisection is used by the well model for thp computations. Maybe the change of the tolerance here also benefits the well models?

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
Looking at the test failure here, it looks like only a single case failed, with small deviations, in other words the usual ok-to-update-data situation. However, at the end of the log was this:
```
Comparing summary files 

Checking 178  vectors  ... Program threw an exception: [/var/lib/jenkins/post-builder/workspace/opm-common-PR-builder/test_util/EclRegressionTest.cpp:964] 
Keyword BGSAT:11,13,9 summary vector of different length (24 != 26)
```

Any idea what is going wrong here, @bska?

**Comment by bska** [comment]
> However, at the end of the log was this:
> ```
> Comparing summary files 
> [ ... ]
> summary vector of different length (24 != 26)
> ```
> Any idea what is going wrong here, @bska?

Sort of.  The direct cause of that message is that one simulation ended up taking more timesteps than the other.  Depending on which of the two values (24 or 26) refers to the current reference and which refers to the results of this PR we either took two fewer or two more timesteps compared to the reference.

There may be any number of reasons for that to happen.

**Comment by atgeirr** [comment]
> we either took two fewer or two more timesteps compared to the reference.

Thanks, I'll investigate.

**Comment by bska** [comment]
> > we either took two fewer or two more timesteps compared to the reference.
> 
> Thanks, I'll investigate.

Cool.  I also had a quick look at the transcript and at least in this case it was the new simulation that took 26 timesteps.  The reference case used 24 timesteps.

**Comment by atgeirr** [comment]
As far as I can tell, the failing test behaves as follows:
| Code | My machine |Jenkins |
| ---- | ------------- | ------------- |
| Master | Chops 3 times  | Does not chop  |
| This PR | Does not chop  | Chops 2 times  |

This test is clearly too unstable to be considered converged. I therefore think we should do a data_update and merge this PR. I'll start the data_update, but please voice your concerns, if any.

**Comment by atgeirr** [comment]
jenkins build this update_data please

**Comment by atgeirr** [comment]
jenkins build this update_data please

**Comment by alfbr** [comment]
> This test is clearly too unstable to be considered converged.

Which test is this?

**Comment by atgeirr** [comment]
> Which test is this?

compareECLFiles_flow+9_3E_GAS_MIN_EXPORT_MSW

**Comment by atgeirr** [comment]
jenkins build this update_data please

**Comment by bska** [comment]
jenkins build this opm-tests=323 please

**Pull request merged**